apps/api/src/profiles/profiles.service.ts

```diff
@@ -1,27 +1,68 @@
 import { Injectable } from '@nestjs/common';
+import { PrismaClientUnknownRequestError } from '@prisma/client/runtime';
+import { HandlerService } from 'src/handler/handler.service';
 import { CreateProfileDto } from './dto/create-profile.dto';
+import { PrismaService } from '../prisma/prisma.service';
 import { UpdateProfileDto } from './dto/update-profile.dto';
-import { Profile } from './entities/profile.entity';
 
 @Injectable()
 export class ProfilesService {
-  create(createProfileDto: CreateProfileDto): Profile {
-    return { name: createProfileDto.name };
-  }
-
-  findAll(): Profile[] {
-    return [{ name: 'test' }];
-  }
+  constructor(private prismaService: PrismaService, private ResponseHandler: HandlerService) {}
 
-  findOne(id: number): Profile {
-    return { name: `test ${id}` };
+  // Method to CREATE a new profile
+  async create(createProfileDto: CreateProfileDto) {
+    try {
+      const resp = await this.prismaService.user.create({
+        data: createProfileDto,
+      });
+      return this.ResponseHandler.Success({ data: resp, message: 'User was created succesfully' });
+    } catch (err) {
+      if (err instanceof PrismaClientUnknownRequestError) {
+        throw err;
+      } else {
+        return this.ResponseHandler.Fail({ message: 'Couldnt create user' });
+      }
+    }
   }
 
-  update(id: number, updateProfileDto: UpdateProfileDto) {
-    return { name: `${updateProfileDto.name} ${id}` };
+  // Method to READ an existing profile
+  async read(id: number) {
+    try {
+      const resp = await this.prismaService.user.findUnique({
+        where: {
+          id,
+        },
+      });
+      return this.ResponseHandler.Success({
```

Comment:
The response generic can be achived by using exception filter and interceptor

https://docs.nestjs.com/exception-filters
https://peaku.co/questions/18266-%C2%BFcomo-crear-un-formato-de-respuesta-api-de-nest-js-comun-para-cada-api